tests/integration/kubernetes/tests_common.sh

```diff
@@ -100,17 +87,16 @@ exec_host() {
 	local command="${@:2}"
 	command+='; echo -en \\n$?'
 
-	# Get the already existing debugger pods
-	local old_debugger_pods=($(kubectl get pods -o name | grep node-debugger))
-
 	# Run a debug pod
-	NODE_NAME="${node}" envsubst < runtimeclass_workloads/custom-node-debugger.yaml | kubectl apply -f - > /dev/null
-
-	# Identify the new debugger pod
-	local new_debugger_pod=$(get_new_debugger_pod "${old_debugger_pods[@]}")
+	# Check if there is a debugger pod named `custom-node-debugger` and reuse it
+	# Otherwise, create a new one
+	if ! kubectl get pod -n kube-system custom-node-debugger > /dev/null 2>&1; then
```

Comment:
Hi @BbolroC ! I really like this idea of keeping the debugger pod alive throughout the tests lifecycle, however, on multi-nodes cluster it is a bit more challenging.

Here suppose that `custom-node-debugger` exists and was created for the node A. Then a test want to `exec_host` on the node B, it will end up running the command on A :(

I see two ways of solving that:

1. Create the pod with  `custom-node-debugger-${NODE_NAME_HASH}`
    At the test suite exit, you will need to find and delete all  `custom-node-debugger-${NODE_NAME_HASH}`
1. Create a DaemonSet that will span  `custom-node-debugger` through all nodes
   At the test suite exit, you only need to delete the daemonset (that the pods will be deleted)
   The tricky part of this approach is to figure out which pod was scheduled to each node so that you can `kubectl exec` on the right pod.

Reply:
yeah, the 1st one sounds better. I will update the PR. Thanks for the feedback. 😉